### ISSUE TITLE
Getting BYTERANGE attribute for #EXT-X-MAP  

### ISSUE THREAD

**Issue by Anjalivishnoi** [globocom/m3u8#278]
#EXT-X-MAP tag consists of an optional attribute BYTERANGE for cases where initialization segments are not independent segments and are part of some other segment. 
Currently we can only get uri for this field via parsed_m3u8.segment_map['uri']. 
There are pretty common usecases where BYTERANGE would be specified as part of EXT-X-MAP specially for CMAF content.
Ask with this issue is to add support for the same.

**Comment by davemevans** [comment]
Map byterange should already be available where the attribute is present using `parsed_m3u8.segment_map['byterange']`. This can be shown by applying the following patch, after which the test will still pass:

```diff --git a/tests/test_parser.py b/tests/test_parser.py
index 0047aaa..9f463d0 100644
--- a/tests/test_parser.py
+++ b/tests/test_parser.py
@@ -332,6 +332,7 @@ def test_should_parse_segment_map_uri():
 def test_should_parse_segment_map_uri_with_byterange():
     data = m3u8.parse(playlists.MAP_URI_PLAYLIST_WITH_BYTERANGE)
     assert data['segment_map']['uri'] == "main.mp4"
+    assert data['segment_map']['byterange'] == '"812@0"'

```
I agree it does kind of suck that the quotes are not removed as normal with quoted-string attributes. This would be a simple one-liner change in the parser and model to acheive this.

```diff --git a/m3u8/model.py b/m3u8/model.py
index 6182b2e..3adbf2f 100644
--- a/m3u8/model.py
+++ b/m3u8/model.py
@@ -755,7 +755,7 @@ class InitializationSection(BasePathMixin):
         if self.uri:
             output.append('URI=' + quoted(self.uri))
         if self.byterange:
-            output.append('BYTERANGE=' + self.byterange)
+            output.append('BYTERANGE=' + quoted(self.byterange))
         return "{tag}:{attributes}".format(tag=self.tag, attributes=",".join(output))

     def __eq__(self, other):
diff --git a/m3u8/parser.py b/m3u8/parser.py
index 07a1032..2df44f0 100644
--- a/m3u8/parser.py
+++ b/m3u8/parser.py
@@ -159,7 +159,7 @@ def parse(content, strict=False, custom_tags_parser=None):
             data['is_endlist'] = True

         elif line.startswith(protocol.ext_x_map):
-            quoted_parser = remove_quotes_parser('uri')
+            quoted_parser = remove_quotes_parser('uri', 'byterange')
             segment_map_info = _parse_attribute_list(protocol.ext_x_map, line, quoted_parser)
             state['current_segment_map'] = segment_map_info
             # left for backward compatibility
diff --git a/tests/test_parser.py b/tests/test_parser.py
index 0047aaa..33e215a 100644
--- a/tests/test_parser.py
+++ b/tests/test_parser.py
@@ -332,6 +332,7 @@ def test_should_parse_segment_map_uri():
 def test_should_parse_segment_map_uri_with_byterange():
     data = m3u8.parse(playlists.MAP_URI_PLAYLIST_WITH_BYTERANGE)
     assert data['segment_map']['uri'] == "main.mp4"
+    assert data['segment_map']['byterange'] == "812@0"


 def test_should_parse_multiple_map_attributes():
```

**Comment by mauricioabreu** [comment]
@Anjalivishnoi can you paste your HLS playlist here?

**Comment by Anjalivishnoi** [comment]
Basic Sample manifest which could be used to test this Issue :

`#EXTM3U
#EXT-X-TARGETDURATION:6
#EXT-X-VERSION:7
#EXT-X-MEDIA-SEQUENCE:1
#EXT-X-PLAYLIST-TYPE:VOD
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-MAP:URI="main.mp4",BYTERANGE="719@0"
#EXTINF:6.00000,	
#EXT-X-BYTERANGE:1508000@719
main.mp4
#EXTINF:6.00000,	
#EXT-X-BYTERANGE:1510244@1508719
main.mp4
#EXTINF:6.00000,	
#EXT-X-BYTERANGE:1504861@3018963
main.mp4
#EXTINF:6.00000,	
#EXT-X-BYTERANGE:1504227@4523824
main.mp4
#EXT-X-ENDLIST
`

**Comment by mauricioabreu** [comment]
I just opened a new pull request https://github.com/globocom/m3u8/pull/279

It will solve both #276 and this #278

**Comment by mauricioabreu** [comment]
Fixed via #279
